### PR TITLE
CAM-13345: preserve relative path when switching version

### PR DESCRIPTION
**Pull request by ThorbenLindhauer**
related to CAM-13345

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -345,11 +345,26 @@ metaHeader.addEventListener('click', toggleMenuMeta);
...
+
+    var dest = attr(versionSelect, 'data-basepath') + '/' + versionSelect.value;  
+
+    var basePath = attr(versionSelect, 'data-basepath');
+    var pathRegex = new RegExp('^' + escapeRegExp(basePath) + '\\/[\\w\\.]+\\/(?<relativePath>.*)');
+
+    var match = pathRegex.exec(location.pathname);
+    if (match !== null) {
+      dest = dest + '/' + match.groups.relativePath;
+    }
```
Comment (marstamm) [inline scripts/index.js:357]: While the Regex works, I think it is a bit overkill to get the site path.

I would add the baseUrl to the select element, which gives us something like `/manual/7.14/`, in https://github.com/camunda/camunda-docs-theme/blob/master/layouts/partials/menu.html#L47 : `base-url="{{ $.Site.BaseURL }}"`

With this, splitting becomes much easier 
```suggestion
    var basePath = attr(versionSelect, 'base-url');
    var result = location.href.split(basePath)[1];
    if (result) {
      dest = dest + '/' + result;
    }
```

**Review by marstamm — CHANGES_REQUESTED** [review]
Nice work, I like the approach :+1:

I think it can be simplified a bit if we use the hugo site variables https://gohugo.io/variables/site/

File: scripts/index.js (hunk 1 above)
Comment (ThorbenLindhauer) [reply scripts/index.js:357]: Thanks, I'll give it a try. I agree it's much better without a regex, but couldn't quite make the connection that this code can easily access the version of the docs it is executed in (or the full base url for that matter).

**Review by marstamm — APPROVED** [review]
The component looks good, unfortunately I noticed problems on the live page.

Currently, the rewrite only works for pages that are also available in the older version. For example, when you are viewing the `User Guide>Camunda Run` Page and change the version to **7.12**, you will end up on the start page of **7.14**, which is the default for pages that are not found.

I will check if I can find an easy way to change this behavior. We should not merge this before it works end to end.

**Pull request merged**